### DISCUSSION TITLE
About setting Dns server

### DISCUSSION THREAD

**Discussion by gjroots** [JFC-Group/JF-Customisation#43] [Q&A] [answered]
Is there a solution or script available to ensure that custom DNS addresses set in the router remain in effect after rebooting the router?

**Comment by itsyourap** [comment]
We can run a script every time the router starts. The script will run a few seconds after the router indicator turns green.
If you want I can write a guide on how to do it.

**Reply by gjroots** [reply]
Ohh. Nice happy to hear. Please tell me, I am  already Accessed shell via telenet.

**Reply by itsyourap** [reply, marked as answer]
File location: `/flash2/pfrm2.0/etc/customInit`

```
#!/bin/sh

# /flash2/pfrm2.0/etc/customInit

echo -e "password\npassword" | passwd root
/usr/sbin/telnetd
/pfrm2.0/bin/iptables -I fwInBypass -p tcp --dport 23 -m ifgroup --ifgroup-in 0x1/0x1 -j ACCEPT
```

This script actually executes on every router reboot. You can modify this script to do whatever you want.

_P.S. - Just make sure the script is in LF and the file has +x permission_

**Reply by gjroots** [reply]
Ok, so the router using dnsmasq so updating config file whenever it after reboots change nameserver in file. Using this script technique.

**Reply by itsyourap** [reply]
Yes

**Reply by gjroots** [reply]
Ok, but if ONT stuck at red light even though rest will also not work, like wifi stops, ethernet also not getting IP, Then this condition happened how to resolve, I tried but boot setting has login page so, I raise a complaint and replaced new one, but real time how to solve you face anything like that? 

In last time I replaced system.db using ftp Filezilla software, on that time it will crashed and stay red light.

**Reply by itsyourap** [reply]
We generally make sure that the script works before directly running it on every reboot.
We do that by first testing the script once after reboot. I mean we add the line to check if the script has run previously (by setting a flag in the persistent `/flash2`) which if true, we do not execute the script again. 

We do so to prevent any misconfigurations that might brick the router.

**Reply by gjroots** [reply]
Yes, it's correct. Do you find any solution to flash firmware if it corrupted.

**Comment by gjroots** [comment]
Hi, I am successfully done. 
I am directly changed firmware code in Lua file then it works for me. Now even after rebooting setting will working.

**Reply by vaibhavramdasi** [reply]
can you share the steps and code?

**Reply by gjroots** [reply]
follow  given instruction below.

**Comment by gjroots** [comment]
Hi,

### Follow these instructions to **SET DNS Server Settings**

1. First change  DNS settings as you want from the admin page and save. 
2. Navigate to **Administration » Maintenance » Backup / Restore** and create a backup of your settings.
3. Obtain root permission as instructed by the Author.

### Now, the real work begins:

1. After gaining root access, enter the following command:
   ```
   cd /usr/local/share/lua/5.1/teamf1lualib
   ```
2. Locate and modify the following two files:
   - **IPv4**: `dhcpc.lua`
   - **IPv6**: `dhcpv6.lua`
3. Edit the specified lines in each file:
   - For **dhcpc.lua**, find and change:
     ```
     local USE_DNS_PROXY = "3"
     ```
   - For **dhcpv6.lua**, find and change:
     ```
     v["dhcpv6s.useDNSServersFrom"] = "3"
     ```
5. To edit these files, use the VI editor. Enter:
   ```
   vi dhcpc.lua
   ```
   and
   ```
   vi dhcpv6.lua
   ```
   Make sure you are familiar with the basics of using the VI editor.



 ### Disclaimer
_This is Only for educational purposes. No one is responsible for any type of damage._

**Reply by vaibhavsabharwal-97** [reply]
dhcpc.lua is read only

**Reply by gjroots** [reply]
How you edited? How you confirm its read only

**Reply by vaibhavsabharwal-97** [reply]
![Screenshot 2024-09-06 155857](https://github.com/user-attachments/assets/d945c614-6c05-4763-a438-3fecbd9d102d)

**Reply by gjroots** [reply]
Ok no issues, so need to learn some basics about vi editor so press `i` for edit mode and esc and `:q!`   for without saving quit and `:wq` for saving file 

So please learn more about vi editor before do anything.

**Reply by vaibhavsabharwal-97** [reply]
![Screenshot 2024-09-06 161208](https://github.com/user-attachments/assets/c2224f76-7e80-42fb-b8fa-96e1e65bc091)
![Screenshot 2024-09-06 161214](https://github.com/user-attachments/assets/49bbc7bb-857f-4995-a7b8-ce86358ca8f2)

i know how to edit and save the file 😊

**Reply by gjroots** [reply]
Ipv6 file working or not only this file not working? And are you in  root shell or other?

**Reply by vaibhavsabharwal-97** [reply]
both files read only

**Reply by gjroots** [reply]
Ok, I will check and update you soon i am not facing this issue 

You are in ` /usr/local/share/lua/5.1/teamf1lualib` directory or any other?

**Reply by vaibhavsabharwal-97** [reply]
yes i am in /usr/local/share/lua/5.1/teamf1lualib directory

**Reply by gjroots** [reply]
Are you root  user or not?

**Reply by vaibhavsabharwal-97** [reply]
root

**Reply by itsyourap** [reply]
Yeah that file is meant to be read-only.
I can confirm it is read-only in squashfs filesystems (JCOW414, JCO4032).


@gjroots is most probably on a ubifs filesystem which may be writable.

**Reply by gjroots** [reply]
> Yeah that file is meant to be read-only. I can confirm it is read-only in squashfs filesystems (JCOW414, JCO4032).
> 
> @gjroots is most probably on a ubifs filesystem which may be writable.

Hi my model: JCOW411 so it works for me

**Reply by ghost** [reply]
![image](https://github.com/user-attachments/assets/13ff5dd8-960f-4a2f-8267-44e4bf9f2ed5)
same here read only

4032 2.57
